registry/storage/driver/rados/rados.go

```diff
@@ -506,7 +514,7 @@ func (d *driver) generateOid() string {
 
 // Reference a object and its hierarchy
 func (d *driver) putOid(objectPath string, oid string) error {
-	directory := path.Dir(objectPath)
+	directory := path.Join(d.rootdirectory, path.Dir(objectPath))
```

Comment:
This will help with the recursion. But actually increases the failures. *sights*

```diff
diff --git a/registry/storage/driver/rados/rados.go b/registry/storage/driver/rados/rados.go
index 4c4b06c..aee1065 100644
--- a/registry/storage/driver/rados/rados.go
+++ b/registry/storage/driver/rados/rados.go
@@ -539,7 +539,7 @@ func (d *driver) putOid(objectPath string, oid string) error {
 
        // Esure parent virtual directories
        if createParentReference {
-               return d.putOid(directory, "")
+               return d.putOid(path.Dir(objectPath), "")
        }
 
        return nil
```